src/auth/client-authentication.ts

```diff
@@ -36,6 +36,9 @@ export const addClientAuthentication = async ({
   clientSecret,
 }: AddClientAuthenticationOptions): Promise<Record<string, unknown>> => {
   const cid = payload.client_id || clientId;
+  if (payload.authorization_details) {
+    payload.authorization_details = JSON.stringify(payload.authorization_details);
+  }
```

Comment:
Why is this done in the `addClientAuthentication` ? I am not sure I understand why it belongs here, can you elaborate?

Reply:
addClientAuthentication is to add clientAuthentication (either secret or private key jwt), not add additional payload. So yeah, moving to the par method is what we want.

Reply:
As stated in an previous comment, since node-auth0 is a low level SDK, we will directly request the customer to supply AuthorizationDetails in required format. We will avoid making required modification from our end for a low level sdk.